### DISCUSSION TITLE
Warning when trying to use a plugin

### DISCUSSION THREAD

**Discussion by lvclark** [nextflow-io/nextflow#1995] [Q&A] [answered]
I am using Nextflow 21.03.0-edge from a Linux cluster on my campus, and am trying to test out the [Azure Batch](https://www.nextflow.io/docs/edge/azure.html) functionality so I can write up documentation for a project I'm on.  I'm getting the error:

```
WARN: Nextflow self-contained distribution only allows default plugins -- User config plugins will be ignored: nf-azure
Cannot a find a file system provider for scheme: az
```

I assume that there is something we need to change about the way Nextflow is installed on our cluster in order to allow it to use plugins.  I haven't been able to find any documentation on this topic.

**Comment by pditommaso** [comment]
Install it as described here https://www.nextflow.io/docs/latest/getstarted.html#installation

**Reply by lvclark** [reply]
Could the problem be because I'm loading it using `module load`?  I don't think our IT guys really want us downloading and running executable binaries from our user directories.

**Comment by abhi18av** [comment, marked as answer]
Hi @lvclark ,

This problem might be there due to the `nextflow-21.03.0-edge-all` binary, installed on your cluster. For downloading plugins on the fly it's better to rely on the usual `nextflow` binary, both of which could be found here https://github.com/nextflow-io/nextflow/releases/tag/v21.03.0-edge

**Reply by lvclark** [reply]
Turns out we did have edge-all installed!  The IT guy reinstalled according to your directions and I'm not getting the plugin warning any more.  Still getting the error about not having an `az` file system though.  And I think at https://www.nextflow.io/docs/edge/azure.html there may be a typo where `id 'nf-azure` should be `id = 'nf-azure`.

**Reply by lvclark** [reply]
Or wait, no I guess `id 'nf-azure` was at least recognized as indicating a plugin in our previous installation.  But now it gives me a Java error (`No signature of method: nextflow.config.ComposedConfigSlurper$_parse_closure5.id() is applicable for argument types: (java.lang.String) values: [nf-azure]`) if I don't use the equals sign.

**Reply by lvclark** [reply]
Never mind, I see that the version of Nextflow I have loaded is super old and is not the one I thought it was.  I'll troubleshoot more before I continue to bug you.

**Reply by lvclark** [reply]
Ok, got the right version of Nextflow, got the nf-azure plugin, and it's working.  Thank you for your help!
